src/main/java/org/xolstice/maven/plugin/protobuf/AbstractProtocMojo.java

```diff
@@ -833,8 +834,7 @@ protected void doAttachFiles() {
                 while (jarEntries.hasMoreElements()) {
                     final JarEntry jarEntry = jarEntries.nextElement();
                     final String jarEntryName = jarEntry.getName();
-                    // TODO try using org.codehaus.plexus.util.SelectorUtils.matchPath() with DEFAULT_INCLUDES
-                    if (jarEntryName.endsWith(PROTO_FILE_SUFFIX)) {
+                    if (isIncludedFile(jarEntryName)) {
```

Comment:
There's no need for that actually. All we need is just:
```
if (SelectorUtils.matchPath(DEFAULT_INCLUDES, jarEntryName)) {
    ...
}
```
Then we can change `DEFAULT_INCLUDES` to something like `"**/*.proto?"` and delete `PROTO_FILE_SUFFIX`.

There are two places where the plugin scans for protobuf source files:
1. When looking for protobuf sources in the current project. In this case it uses configurable `includes` and `excludes` properties, where `includes` defaults to the `DEFAULT_INCLUDES` and `excludes` defaults to nothing.
2. When looking for protobuf sources in the dependencies (those can be either jars or directories). In that case it simply applies `DEFAULT_INCLUDES`. This is the case in this particular line of code.

The check in `Protoc.addProtoFile()` is somewhat redundant in a larger context, because the directory/jar scanners guarantee that only `*.proto` files are selected.